src/jit/block.h

```diff
@@ -144,6 +144,85 @@ struct EntryState
     StackEntry* esStack;               // ptr to  stack
 };
 
+// Enumeration of the kinds of memory whose state changes the compiler tracks
+enum MemoryKind
+{
+    ByrefExposed = 0, // Includes anything byrefs can read/write (everything in GcHeap, address-taken locals,
+                      //                                          unmanaged heap, callers' locals, etc.)
+    GcHeap,           // Includes actual GC heap, and also static fields
+    MemoryKindCount,  // Number of MemoryKinds
+};
+#ifdef DEBUG
+const char* const memoryKindNames[] = {"ByrefExposed", "GcHeap"};
+#endif // DEBUG
+
+// Bitmask describing a set of memory kinds (usable in bitfields)
+typedef unsigned int MemoryKindSet;
+
+// Bitmask for an empty MemoryKindSet
+inline MemoryKindSet memoryKindSet()
+{
+    return 0;
+}
+
+// Bitmask for a MemoryKindSet containing the specified MemoryKinds
+template <typename... MemoryKinds>
+inline MemoryKindSet memoryKindSet(MemoryKind id, MemoryKinds... ids)
+{
+    return (1U << id) | memoryKindSet(ids...);
+}
```

Comment:
I'm not familiar with this syntax:
`typename...`  `MemoryKinds...' and `ids...`
What is this called?
And is it supported across our compilers?